.buildkite/commands/set-up-git-to-fetch-wordpress-rs.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -eu
+
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
+  echo "This script must be sourced, not executed, because it exports GIT_SSH_COMMAND."
+  exit 1
+fi
+
+echo "--- :git: Change Git SSH key to fetch private dependencies"
+
+WORDPRESS_RS_FETCH_KEY_NAME="read_only_deploy_key_to_fetch_wordpress_rs"
+# $WORDPRESS_RS_KEY is declared in the `env` file of this pipeline in `mobile-secrets`
+add_ssh_key_to_agent "$WORDPRESS_RS_KEY" "$WORDPRESS_RS_FETCH_KEY_NAME"
+WORDPRESS_RS_FETCH_KEY="$HOME/.ssh/$WORDPRESS_RS_FETCH_KEY_NAME"
+
+export GIT_SSH_COMMAND="ssh -i $WORDPRESS_RS_FETCH_KEY -o IdentitiesOnly=yes"
+echo "Git SSH command is now $GIT_SSH_COMMAND"
```

Comment:
This is quite misleading because it blanket-updates the key for all operations. It works right now because the only private repo we have is WordPress-rs. 

Given there is no plan (?) to add more private repos and in fact WordPress-rs will soon™ become public, I can live with how suboptimal this is. After all, it should be short lived.

Reply:
For the record, even if we wanted `ssh` to still test other keys and not _only_ use that deploy key, we can drop the `-o IdentitiesOnly=yes`. So if we needed to support accessing more private repos one day, be it here in WPiOS or in other repos, this should still be doable.

 - `ssh -i mykey1`: SSH will add that key to the list of keys also provided by the `ssh-agent` and by the `~/.ssh/config`. When it will try to connect, it will try the keys in turn, until it find ones that the server accepts
 - `ssh -i mykey1 -o IdentitiesOnly=yes`: Adds that key to the list of keys it knows, but also tells `ssh` to not use the keys provided by `ssh-agent` when it will try all the keys it knows when it tries to connect. It will still account for any identity declared in `~/.ssh/config` though (if any fits), in addition to the `mykey1` you provided.
 - `ssh -i mykey1 -o IdentitiesOnly -F /dev/null`: Adds `mykey1` to the list of known keys, but also ignores keys provided by the `ssh-agent`, and also ignore keys declared in `~/.ssh/config` (by telling it to use `/dev/null` as the config file instead). Which means that in the end `mykey1` is the only one which will be tried
 - `ssh -i key-for-private-repo-1 -i key-for-private-repo-2 -i key-for-private-repo-3`: SSH will try the keys known to the `ssh-agent` first (especially the `~/.mobile-secrets/CI/secrets/wordpress-ios/private_ssh_key` which was [installed in the SSH agent by Buildkite in the agent via `s3-secrets-buildkite-plugin`](https://github.com/buildkite/elastic-ci-stack-s3-secrets-hooks)), which will work for `git fetch` operations that are for a public repo for example, but fail for private repos. Then if that key fails, it will try keys in `~/.ssh/config` (likely there aren't any defined there on our CI agents), and finally try `key-for-private-repo-1`, then `…-2`, then `…-3` in turn until it finds one that is accepted.

That last approach means that we can still add more read-only deploy keys to the SSH command if we were to need to access more private repos, by creating a deploy key for each repo and adding all of them via as many `-i` parameters as needed so that it can try them all and reach all necessary repos.

Reply:
Bookmarking this for reference if we'll ever go down that path on other projects.

Thanks @AliSoftware